src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -0,0 +1,7757 @@
+            }
+        }
+        else
+        {
+            assert(!data->isContained());
+            dataReg = data->GetRegNum();
+        }
+        assert(dataReg != REG_NA);
+
+        if (targetReg == REG_NA) // store into stack based LclVar
+        {
+            inst_set_SV_var(lclNode);
+
+            instruction ins  = ins_StoreFromSrc(dataReg, targetType);
+            emitAttr    attr = emitActualTypeSize(targetType);
+
+            emit->emitIns_S_R(ins, attr, dataReg, varNum, /* offset */ 0);
+
+            genUpdateLife(lclNode);
+
+            varDsc->SetRegNum(REG_STK);
+        }
+        else // store into register (i.e move into register)
+        {
+            if (dataReg != targetReg)
+            {
+                // Assign into targetReg when dataReg (from op1) is not the same register
+                inst_Mov(targetType, targetReg, dataReg, true, emitActualTypeSize(targetType));
+            }
+            genProduceReg(lclNode);
+        }
+    }
+}
+
+void CodeGen::genSimpleReturn(GenTree* treeNode)
+{
+    assert(treeNode->OperGet() == GT_RETURN || treeNode->OperGet() == GT_RETFILT);
+    GenTree*  op1        = treeNode->gtGetOp1();
+    var_types targetType = treeNode->TypeGet();
+
+    assert(targetType != TYP_STRUCT);
+    assert(targetType != TYP_VOID);
+
+    regNumber retReg = varTypeUsesFloatArgReg(treeNode) ? REG_FLOATRET : REG_INTRET;
+
+    bool movRequired = (op1->GetRegNum() != retReg);
+
+    if (!movRequired)
+    {
+        if (op1->OperGet() == GT_LCL_VAR)
+        {
+            GenTreeLclVarCommon* lcl            = op1->AsLclVarCommon();
+            bool                 isRegCandidate = compiler->lvaTable[lcl->GetLclNum()].lvIsRegCandidate();
+            if (isRegCandidate && ((op1->gtFlags & GTF_SPILLED) == 0))
+            {
+                // We may need to generate a zero-extending mov instruction to load the value from this GT_LCL_VAR
+
+                unsigned   lclNum  = lcl->GetLclNum();
+                LclVarDsc* varDsc  = &(compiler->lvaTable[lclNum]);
+                var_types  op1Type = genActualType(op1->TypeGet());
+                var_types  lclType = genActualType(varDsc->TypeGet());
+
+                if (genTypeSize(op1Type) < genTypeSize(lclType))
+                {
+                    movRequired = true;
+                }
+            }
+        }
+    }
+    if (movRequired)
+    {
+        emitAttr attr = emitActualTypeSize(targetType);
+        if (varTypeUsesFloatArgReg(treeNode))
+        {
+            if (attr == EA_4BYTE)
+            {
+                GetEmitter()->emitIns_R_R_R(INS_fsgnj_s, attr, retReg, op1->GetRegNum(), op1->GetRegNum());
+            }
+            else
+            {
+                GetEmitter()->emitIns_R_R_R(INS_fsgnj_d, attr, retReg, op1->GetRegNum(), op1->GetRegNum());
+            }
+        }
+        else
+        {
+            if (attr == EA_4BYTE)
+            {
+                if ((treeNode->gtFlags & GTF_UNSIGNED) != 0)
```

Comment:
I'm confused here, when is `GTF_UNSIGNED` ever set on a `GT_RETURN` node?

Reply:
Removed.